Assets/UI Toolkit/Scripts/Components/Icon.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Netherlands3D.UI;
+using Netherlands3D.UI.ExtensionMethods;
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Netherlands3D.UI.Components
+{
+    [UxmlElement]
+    public partial class Icon : VisualElement
+    {
+        public enum IconImage
+        {
+            Plus,
+            Map,
+            Folder,
+            Trash,
+        }
+
+        public enum IconColor
+        {
+            White,
+            Black,
+            Blue50,
+            Blue100,
+            Blue200,
+            Blue300,
+            Blue700,
+            Blue900,
+        }
+
+        private IconImage image = IconImage.Plus;
+        [UxmlAttribute("image")]
+        public IconImage Image
```

Comment:
This declares the new field in the UI Builder named "image" and the `SetFieldValueAndReplaceClassName` method will make this remember the setting - and replace the classname that starts with `image-` so that the image is swapped.

This is a powerful technique where you declare the styling in the USS, and all you do is add and swap classes on it. This decouples design from implementation